src/java/org/apache/cassandra/io/sstable/format/Version.java

```diff
@@ -141,4 +141,11 @@ public int hashCode()
     public abstract boolean hasMaxColumnValueLengths();
 
     public abstract boolean hasOriginatingHostId();
+
+    /**
+     * Whether we expect that sstable has explicitly frozen tuples in its {@link org.apache.cassandra.db.SerializationHeader}.
+     * If {@code true}, we don't try to fix non-frozen tuples that are not types of dropped columns and fail loading
+     * the sstable. If {@code false}, we try to fix non-frozen tuples and load the sstable.
+     */
+    public abstract boolean hasExplicitlyFrozenTuples();
```

Comment:
We always check this negated, three times, so we could replace it by a `hasImplicitlyFrozenTuples` method that returns the opposite: https://github.com/adelapena/cassandra/commit/9357bc73ce685bb7867b3135fc64e46086804529